virtcontainers/types/asset.go

```diff
@@ -18,6 +18,40 @@ import (
 // AssetType describe a type of assets.
 type AssetType string
 
+// AssetTypes returns a list of all known asset types.
+//
+// XXX: New asset types *MUST* be added here.
+func AssetTypes() []AssetType {
+	return []AssetType{
+		FirmwareAsset,
+		HypervisorAsset,
+		HypervisorCtlAsset,
+		ImageAsset,
+		InitrdAsset,
+		JailerAsset,
+		KernelAsset,
+	}
+}
+
+// AssetAnnotations returns all annotations for all asset types.
+func AssetAnnotations() ([]string, error) {
+	var result []string
+
+	for _, at := range AssetTypes() {
+		aPath, aHash, err := at.Annotations()
+		if err != nil {
+			return []string{}, err
+		}
+
+		result = append(result, []string{aPath, aHash}...)
+	}
+
+	// Add the odd one out; it isn't part of a (path+hash) pair.
+	result = append(result, annotations.AssetHashType)
```

Comment:
We dont need to address this in this PR. 
But raising my concern about value of supporting this. Maybe we should just support sha512 hash and document it so, rather than providing this.

Reply:
Fixed.